src/annotated_image.cpp

```diff
@@ -252,7 +252,7 @@ void findAnnotatedROIs(const AnnotatedImage& img, std::vector<ed::EntityConstPtr
 
         for(ed::ImageMask::const_iterator it = mask.begin(rgb.cols); it != mask.end(); ++it)
         {
-            const cv::Point2i& p = *it;
+            const cv::Point2i& p = it();
```

Comment:
:O  So more uses were hidden in other repositories? :)

Can't have a reference here, since `it()` needs storage for its result. Not sure, but it could be that c++ creates anonymous storage here. Even then, the better solution is
`const cv::Point2i p(it());` (ie copy-constructor call, as you can't assign a `const` variable.)

Same problem happens in other points that you found as well.